### ISSUE TITLE
Install instructions for Debian 10

### ISSUE THREAD

**Issue by opusz** [rems-project/sail#50]
Hi!
At first, I am a newbie in Ocaml etc. and I try to build SAIL on my Debian 10 with Ocaml 3.05.

I followed the instructions of the INSTALL.md

It stops at Sail itself and says:
_File "sail.ml", line 448, characters 8-18:_
_Error: Unbound module B64_
_Command exited with code 2._

Due to an API break in the module base64, I changed the version of base64 from 3.2.0 to 2.3.0 and try to build it again:
_File "jib/jib_smt.ml", line 774, characters 19-28:_
_Error: Unbound value List.init_
_Command exited with code 2._

Now, I have no idea how to handle it. I guess some API changes causes this errors and make the install instruction be invalid. Any ideas?

**Comment by rmn30** [comment]
Hi!

Unfortunately it looks like those instructions are a little out of date. List.init was introduced in OCaml 4.06 which latest versions of sail now require. You can install this with opam like so:
```
opam switch 4.06.1
```
(we recommend 4.06.1 release due to a bug in 4.06.0)

However the easiest way to install sail and dependencies is to follow the instructions on the wiki to install our opam packages: https://github.com/rems-project/sail/wiki/OPAMInstall

Thanks for reporting. Please let us know how you get on. I'll leave this bug open until we update INSTALL.md .

**Comment by Alasdair** [comment]
I made INSTALL.md a copy of the opam installation instructions from the wiki, and moved the building from source instructions to another wiki page. I think using opam is a more practical method for 99% of use cases rather than building everything from scratch, especially as you can always use opam pin to use the latest github version.

**Comment by marniedunsmore** [comment]
Thanks Alasdair.   I will try the new INSTALL.md today.

**Comment by opusz** [comment]
> Hi!
> 
> Unfortunately it looks like those instructions are a little out of date. List.init was introduced in OCaml 4.06 which latest versions of sail now require. You can install this with opam like so:
> 
> ```
> opam switch 4.06.1
> ```
> 
> (we recommend 4.06.1 release due to a bug in 4.06.0)
> 
> However the easiest way to install sail and dependencies is to follow the instructions on the wiki to install our opam packages: https://github.com/rems-project/sail/wiki/OPAMInstall
> 
> Thanks for reporting. Please let us know how you get on. I'll leave this bug open until we update INSTALL.md .

Installing via opam works without any issues. Thanks

**Comment by marniedunsmore** [comment]
Alasdair, the install worked for me too.  I now have a directory

~/.opam/repo/rems/packages 

with sub directories:

lem
linksem
sail
sail-riscv

I am going to try to run some of the riscv code.  I'll let you know how it goes.

**Comment by marniedunsmore** [comment]
Looks like the only thing in the sail-riscv/sail-riscv.0.1 directory is a file called opam.  Doesn't look like any of the riscv sail source code is there.  Am i missing something?  How do I get the riscv sail source code installed?

**Comment by rmn30** [comment]
The executables I mentioned will be installed in the bin directory of the opam switch.

**Comment by marniedunsmore** [comment]
/.opam/4.06.1/bin  

?

**Comment by marniedunsmore** [comment]
Are the executables in 

~/.opam/4.06.1/bin

?

**Comment by rmn30** [comment]
They should be e.g. riscv_sim_RV64 . If you configured your shell environment with 'eval  $(opam config env)' that directory should be in $PATH.

**Comment by marniedunsmore** [comment]
Working from INSTALL.md, here is the sequence of instructions I entered:

opam switch create ocaml-base-compiler.4.06.1

eval  `opam config env` 

opam repository add rems https://github.com/rems-project/opam-repository.git

sudo apt-get install build-essential libgmp-dev z3
  
path is :  /home/ubuntu/.opam/ocaml-base-compiler.4.06.1/bin:/.cabal/binr:/usr/local/bin:  . . .a

I don't have any directory in my path with 'riscv_sim_RV64'.

Are the commands in INSTALL.md the ones I should be running?

**Comment by rmn30** [comment]
Yes, those instructions are now the ones to use. It sounds like something must have gone wrong with the package install. It may be because we haven't tested as much with opam version 2.
Please could you send me the output of `opam reinstall -v sail-riscv` ?

**Comment by rmn30** [comment]
I've released a new sail-riscv package that will hopefully fix the issues you were having. Maybe you could try `opam update && opam install sail-riscv` again?

**Comment by marniedunsmore** [comment]
I think that worked.  I can see the riscv directories in ~/.opam/ocaml-base-compiler.4.06.1/bin:

riscv_sim_RV32 
riscv_sim_RV64

Here is the full sequence of commands I typed to get this installed:

opam switch create ocaml-base-compiler.4.06.1

eval 'opam config env'

opam repository add rems https://github.com/rems-project/opam-repository.git

sudo apt-get install build-essential libgmp-dev z3

opam update

opam install sail-riscv

eval $(opam env)

**Comment by marniedunsmore** [comment]
Thank you!
